src/delegate.ts

```diff
@@ -0,0 +1,38 @@
+import { Handle } from 'dojo-core/interfaces';
+import on, { ExtensionEvent } from 'dojo-core/on';
+
+export default function delegate(target: HTMLElement, selector: string, type: string, listener: (event: UIEvent) => void): Handle;
+export default function delegate(target: HTMLElement, selector: string, type: ExtensionEvent, listener: (event: UIEvent) => void): Handle;
+export default function delegate(target: HTMLElement, selector: string, type: (string | ExtensionEvent)[], listener: (event: UIEvent) => void): Handle;
+export default function delegate(target: HTMLElement, selector: string, type: any, listener: (event: UIEvent) => void): Handle {
+	function matches(target: EventTarget) {
+		let node = <any> target;
```

Comment:
Why the any cast?  IMO, we should at least comment whenever we do an any cast, because elsewise it looks like we are simply "cheating" to get around strong typing, and we shouldn't do that without explaining ourselves.